lib/facebot.js

```diff
@@ -154,11 +155,41 @@ Facebot.prototype.createFBApi = function(credentials){
            });
 }
 
+// loop that will continue to send the is typing indicator to a channel
+// until we hear back they are not typing, or 10 minutes have past
+Facebot.prototype.typingLoop = function(link){
+    if (! link.is_typing || ((Date.now() / 1000) - link.typing_start_time) > 60*10) //if they were typing for more than 10 minutes lets assume we missed the typing end event
+        return;
+    var us = this;
+    this.ws.send(JSON.stringify({ type: 'typing', channel: link.slack_channel }));
+    fbUtil.sleep(3000).then( ()=> us.typingLoop(link));
```

Comment:
Can we get rid of all this extra added sleep promise stuff and just go with `setInterval( () => this.typingLoop( link ), 3000 )`?